### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by troglobit**
## Description

 - Split `doc/cli/tutorial.md` in multiple files, for use with cli built-in `help` command
   - Also, add more `help` topics: netcalc, tcpdump
 - Extract YANG documentation at build time from `sysrepoctl -l` 
 - Fix config MD5CRYPT warning at defconfig/oldconfig
 - Fix wrong version for `ietf-tcp-common.yang`

## Checklist

Tick *relevant* boxes, this PR is-a or has-a:

- [ ] Bugfix
  - [ ] Regression tests
  - [ ] ChangeLog updates (for next release)
- [X] Feature
  - [ ] YANG model change => revision updated?
  - [ ] Regression tests added?
  - [x] ChangeLog updates (for next release)
  - [X] Documentation added
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (please detail in commit messages)
- [X] Build related changes: minor fix
- [X] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

File: doc/cli/introduction.md [hunk 1]
```diff
@@ -9,21 +10,76 @@ Nevertheless, when it comes to initial deployment and debugging, it
...
+
+The system has three *main datastores* (or files): *factory*, *startup*,
+and *running* that can be managed and inspected using the `copy`,
+`show`, and `configure` commands.  The traditional names used in the CLI
+for these are listed below:
 
  - `factory-config` consists of a set of default configurations, some
    static and others generated per-device, e.g., a unique hostname and
-   number of ports/interfaces.   This file is generated at boot.
+   number of ports/interfaces.   This file is generated at boot and is
```
Comment (wkz) [inline doc/cli/introduction.md:81]: Pretty soon, this won't be true for all systems. Do we either want to mention that, or remove the bit about it being generated altogether? (It does seem like a bit of an implementation detail :man_shrugging:)

File: doc/cli/tcpdump.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+The following section is useful for understanding how to use the tool.
+A section called [Examples](#examples) follows that, which may be what
+you want to scroll down to.
+
+
+### Hardware Overview
+
+Using `tcpdump` effectively requires an understanding of how the
+underlying hardware works.  For a standard PC, or common SBC, the
```
Comment (wkz) [inline doc/cli/tcpdump.md:16]: I'm not sure how widely used the term "Single Board Computer" is, maybe we should expand it?

File: doc/cli/tcpdump.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                    |                    |
+                 ---+                    +---
+                X1  |         SC         |  X3
+                 ---+                    +---
+                X2  |                    |  X4
+                    '--+--+--+--+--+--+--'
+                       |  |  |  |  |  |
+                      X5 X6 X7 X8 X9 X10
+
+So, running `tcpdump x1` in a switching setup, inside the CPU, the only
```
Comment (wkz) [inline doc/cli/tcpdump.md:51]: I think `s/X/e/g` would be more in line with the typical system.

File: doc/cli/tcpdump.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+tcpdump, which means you need an external device (laptop).
+
+> A planned feature is to support mirroring traffic to the CPU port,
+> which would be an effective way to log and monitor traffic over a
+> longer period of time.  Highly effective for diagnosing intermittent
+> and other rare network issues.
+
+If only "proof of life" is required, then sometimes port counters, also
+called *RMON counters*, can be very useful too.  Seeing counters of a
+particular type move means traffic is ingressing or egressing.
```
Comment (wkz) [inline doc/cli/tcpdump.md:66]: `s/move/increment/`, maybe

**Review by wkz — APPROVED** [review]
Förtjusande!

Just found some very minor things in the text that might be worth a look

File: doc/cli/introduction.md (hunk 1 above)
Comment (troglobit) [reply doc/cli/introduction.md:81]: Good point, I'll simplify it, thanks!

File: doc/cli/tcpdump.md (hunk 2 above)
Comment (troglobit) [reply doc/cli/tcpdump.md:16]: Will do, thanks!

File: doc/cli/tcpdump.md (hunk 3 above)
Comment (troglobit) [reply doc/cli/tcpdump.md:51]: OK!

File: doc/cli/tcpdump.md (hunk 4 above)
Comment (troglobit) [reply doc/cli/tcpdump.md:66]: Ah thanks, I was really searching for the correct phrasing!  Fixing ...

File: board/common/post-build.sh [hunk 5]
```diff
@@ -4,6 +4,37 @@ common=$(dirname "$(readlink -f "$0")")
 . "$BR2_CONFIG" 2>/dev/null
 . "$TARGET_DIR/usr/lib/os-release"
 
+# Extract list of loaded YANG modules and their features for yangdoc.html
+mkyangdoc()
+{
+    cmd="yangdoc -o $1 -p $TARGET_DIR/usr/share/yang"
+
+    # shellcheck disable=SC2155
+    export SYSREPO_SHM_PREFIX="foo"
```
[the remark this answers is not in the document]
Comment (troglobit) [reply board/common/post-build.sh:13]: Thanks, was a quick fix that I meant to change, fixing!

File: board/common/post-build.sh [hunk 6]
```diff
@@ -4,6 +4,37 @@ common=$(dirname "$(readlink -f "$0")")
...
+            feature=$(echo "$line" | awk -F'|' '{print $8}' | sed 's/^ *//;s/ *$//')
+            if [ -n "$feature" ]; then
+                feature_list=$(echo "$feature" | tr ' ' '\n')
+                for feat in $feature_list; do
+                    cmd="$cmd -e $feat"
+                done
+            fi
+        fi
+    done <<EOF
+$(SYSREPO_SHM_PREFIX="foo" sysrepoctl -l; rm -f /dev/shm/foo*)
```
[the remark this answers is not in the document]
Comment (troglobit) [reply board/common/post-build.sh:27]: Ah, of course, thanks!

**Review by mattiaswal — APPROVED** [review]

**Pull request merged**